core/shared/src/main/scala-2.12+/clock.scala

```diff
@@ -41,4 +42,10 @@ package object clock extends Clock.Service[Clock] {
    */
   final def sleep(duration: Duration): ZIO[Clock, Nothing, Unit] =
     ZIO.accessM(_.clock sleep duration)
+
+  /**
+   * Returns the java.time.ZoneId corresponding to the System timeZone
+   */
+  val timeZone: ZIO[Clock, Nothing, ZoneId] =
+    ZIO.accessM(_.clock.timeZone)
 }
```

Comment:
Should we also consider introducing `currentDateTime`, as a high-level alternative to `currentTime`?

Something like:
```scala
    /** Get the current time, represented in the current timezone. */
    val currentDateTime: ZIO[Clock, Nothing, OffsetDateTime] = {
      for {
        millis <- currentTime(TimeUnit.MILLISECONDS)
        zone   <- timezone
      } yield OffsetDateTime.ofInstant(Instant.ofEpochMilli(millis), zone)
    }
```

I feel this is what the vast majority of people will be using `timezone` for, so it would be convenient to have it built-in.

Reply:
That would be very useful IMO